server/fleet/windows_mdm.go

```diff
@@ -29,6 +35,48 @@ type MDMWindowsConfigProfile struct {
 	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
 }
 
+// ValidateUserProvided ensures that the SyncML content in the profile is valid
+// for Windows.
+//
+// It checks that all top-level elements are <Replace> and none of the <LocURI>
+// elements within <Target> are reserved URIs.
+//
+// Returns an error if these conditions are not met.
+func (m *MDMWindowsConfigProfile) ValidateUserProvided() error {
+	if mdm.GetRawProfilePlatform(m.SyncML) != "windows" {
+		return errors.New("TODO")
+	}
+
+	doc := etree.NewDocument()
+	if err := doc.ReadFromBytes(m.SyncML); err != nil {
```

Comment:
it does! I also tried wrapping the bytes  in a tag and using the standard library to parse but it was more cumbersome. 

I honestly don't mind switching to the stdlib at all if we think it can make things more clear.

Reply:
Great! I think that's fine like this, we already import this dependency anyway AFAIK so it's not introduced just for that.